signature-aggregator/aggregator/cache/cache.go

```diff
@@ -0,0 +1,54 @@
+package cache
+
+import (
+	"sync"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"go.uber.org/zap"
+)
+
+const (
+	initialCapacity = 1024
+	maxSize         = 1024 * 1024
+)
+
+type Cache struct {
+	logger logging.Logger
+	lock   sync.RWMutex
+	cache  map[[ids.IDLen]byte]warp.Message
+}
+
+func NewCache(logger logging.Logger) *Cache {
+	return &Cache{
+		cache:  make(map[[ids.IDLen]byte]warp.Message, initialCapacity),
+		logger: logger,
+	}
+}
+
+// if there is a cached value, returns the value and true. if there isn't,
+// returns a zero value and false.
+func (c *Cache) Get(msgID ids.ID) (warp.Message, bool) {
+	c.lock.RLock()
+	value, isCached := c.cache[msgID]
+	c.lock.RUnlock()
+
+	if isCached {
+		c.logger.Debug("cache hit", zap.Stringer("msgID", msgID))
+	} else {
+		c.logger.Debug("cache miss", zap.Stringer("msgID", msgID))
+	}
+
+	return value, isCached
+}
+
+func (c *Cache) Set(warpMsgID ids.ID, value *warp.Message) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if len(c.cache) < maxSize {
+		c.cache[warpMsgID] = *value
+	} else {
+		c.logger.Warn("Cache length is at max size; skipped Set")
```

Comment:
Since keys don't expire here we should make this an LRU (or FIFO?) cache.

Reply:
the current commit uses the same LRU cache as the compliance screener.